src/server/game/Instances/InstanceScript.h

```diff
@@ -135,6 +135,22 @@ struct MinionInfo
     BossInfo* bossInfo;
 };
 
+struct InstanceSpawnGroupInfo
+{
+    enum
+    {
+        FLAG_ACTIVATE_SPAWN = 0x01,
+        FLAG_BLOCK_SPAWN = 0x02,
+
+        FLAG_ALL = (FLAG_ACTIVATE_SPAWN | FLAG_BLOCK_SPAWN)
+    };
+    uint8 bossStateId;
+    uint8 bossStates;
+    uint32 spawnGroupId;
+    uint8 flags;
+};
+typedef std::unordered_multimap<uint16, InstanceSpawnGroupInfo> InstanceSpawnGroupMap;
```

Comment:
I still don't like this, none of these headers should include each other (in any direction)
Proposed fix: change the container to `std::unordered_map<uint16, std::vector<InstanceSpawnGroupInfo>>`
and hold a pointer to that vector in InstanceScript